src/App.tsx

```diff
@@ -17,18 +39,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onSetSearch={setSearchFilter}
+                onsetQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0
+                ? (
+                  <TodoList
+                    todos={displayedTodos}
+                    selectedTodoId={selectedTodo?.id}
+                    selectTodo={setSelectedTodo}
+                  />
+                ) : (
+                  <Loader />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo?.id && (
```

Comment:
It's gonna be null when TodoModal is closed
```suggestion
      {selectedTodo && (
```